duckduckgo/Android
# fix: don't add empty domains to whitelist

Task/Issue URL: https://app.asana.com/0/414730916066338/1199127532699902/f
Tech Design URL: 
CC: 

**Description**:

It doesn't seem to have any impact, but I found it odd during debugging that there is an empty domain in the whitelist:
![Screenshot from 2020-10-31 13-59-51](https://user-images.githubusercontent.com/8261416/97788185-ebd81b00-1bae-11eb-8259-a8e5922ba382.png)
This seemed to happen due to empty line at the end of the whitelist, so I added a check to see if the line is not blank before mapping it to the white list domain model.

An alternative solution would be to not allow constructing white list domain models without valid domains in the first place.

A unit test for `downloadTemporaryWhitelist` would be nice :upside_down_face: 

**Steps to test this PR**:
1. open browser, but with empty cache
1. check domains list of whitelist in debugger, no more empty domain added at the end


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
